openshift/WFLY-14936_New_S2I_Image.adoc

```diff
@@ -0,0 +1,204 @@
+user CLI scripts execution, ability to enrich server content and application(s) deployment. 
+
+This plugin used in conjunction with this new WildFly s2i image is all what is needed to offer a flexible experience. 
+We gain the flexibility to provision/configure a WildFly instance to answer specific requirements instead of 
+relying on a fixed set of features packaged inside a WildFly s2i builder image.
+
+N.B.: The image is not strongly coupled with this Maven plugin. For source build, any plugin that outputs a server installation can be used during S2I source build.
+For binary build, any server installation can be the source of binary build.
+
+==== Legacy cloud Galleon feature-pack
+
+All the existing launch scripts and configurability aspects are moved inside a legacy cloud Galleon feature-pack 
+that one can use to build a server to be installed in the new WildFly S2I image.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14936[WFLY-14936]
+
+=== Related Issues
+
+* Server Mode for Bootable JAR: https://issues.redhat.com/browse/WFLY-14934[WFLY-14934]
+
+* JKube evolved support: https://issues.redhat.com/browse/WFLY-14936[WFLY-14938]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[Bootable JAR Maven plugin]
+
+* https://github.com/eclipse/jkube[JKube]
+
+* https://github.com/wildfly/wildfly-s2i[WildFly s2i]
+
+* https://github.com/wildfly/quickstart[WildFly quickstarts]
+
+* https://github.com/shipwright-io/build[Shipwright, S2I build framework]
+
+* ODO dev files.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [ ] Bootable jar
+
+== Requirements
+
+=== Implementation notes
+
+In WildFly s2i repo, a new git branch v2 will be used to implement the feature and release the first versions of the images. The master will be used to 
+build and release the current WildFly S2I and runtime images that will become deprecated. The build and release of the legacy cloud Galleon feature-pack 
+is also done in the master branch.
+Once the deprecated images are no more released, a legacy branch will be created, the v2 branch will be merged into master. At this point, no more "legacy" content will be released.
+
+=== Hard Requirements
+
+==== WildFly s2i builder image
+
+===== Building
+
+* The builder image is built from ubi8-minimal image.
```

Comment:
as the image is runnable, would it make sense to base it instead on `openjdk11` + maven tooling?
This would be quite symmetric with the runtime image:
* WildFly S2I Builder Image -> based on openjdk
* WildfFly S2I Runtime image -> based on openjdk-jre

Reply:
Openjdk builder image has its own implementation of S2I that can't be overridden in a child image.